Firebase/Messaging/FIRMessagingRemoteNotificationsProxy.m

```diff
@@ -23,44 +23,36 @@
 #import "FIRMessagingLogger.h"
 #import "FIRMessagingUtilities.h"
 #import "FIRMessaging_Private.h"
+#import <GoogleUtilities/GULAppDelegateSwizzler.h>
 
-static const BOOL kDefaultAutoRegisterEnabledValue = YES;
 static void * UserNotificationObserverContext = &UserNotificationObserverContext;
 
 static NSString *kUserNotificationWillPresentSelectorString =
     @"userNotificationCenter:willPresentNotification:withCompletionHandler:";
 static NSString *kUserNotificationDidReceiveResponseSelectorString =
     @"userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:";
-static NSString *kReceiveDataMessageSelectorString = @"messaging:didReceiveMessage:";
```

Comment:
I am not sure why this method was swizzled. It is invoked by SDK itself, so it can be called directly if needed. It doesn't look like it is actually needed, because the message is actually processed before the method is called on the messaging delegate.

@chliangGoogle Please let me know if I am missing something.

Reply:
yep, that's for analytics tracking as well.
So any message received, we will trigger to call analytics to log message received related events.

Reply:
Look for "  [[FIRMessaging messaging] appDidReceiveMessage:"